### PR TITLE
fix marianMT convertion to onnx

### LINKED ISSUE #19283: python -m transformers.onnx --model=Helsinki-NLP/opus-mt-en-zh onnx/
### System Info

transformers:4.22.2
python3.8.4
win10   

raise ValueError(
ValueError: Outputs values doesn't match between reference model and ONNX exported model: Got max absolute difference of: 
2.8133392333984375e-05

### Who can help?

_No response_

### Information

- [ ] The official example scripts
- [X] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

 python -m transformers.onnx --model=Helsinki-NLP/opus-mt-en-zh onnx/

### Expected behavior

Export onnx and translate through onnx
https://www.kaggle.com/code/catchlife/translate-opt
Custom export has incorrect translation results

### PR DESCRIPTION
**Pull request by kventinel**
# What does this PR do?

Fixes #19283

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: src/transformers/onnx/convert.py [hunk 1]
```diff
@@ -438,10 +438,12 @@ def validate_model_outputs(
 
         # Values
         if not np.allclose(ref_value, ort_value, atol=atol):
+            bad_idxes = np.logical_not(np.isclose(ref_value, ort_value, atol=atol))
```
Comment (lewtun) [inline src/transformers/onnx/convert.py:441]: nit: we usually prefer being explicit where possible with variable names
```suggestion
            bad_indices = np.logical_not(np.isclose(ref_value, ort_value, atol=atol))
```

File: src/transformers/onnx/convert.py [hunk 2]
```diff
@@ -438,10 +438,12 @@ def validate_model_outputs(
...
             logger.info(f"\t\t-[x] values not close enough (atol: {atol})")
             raise ValueError(
                 "Outputs values doesn't match between reference model and ONNX exported model: "
-                f"Got max absolute difference of: {np.amax(np.abs(ref_value - ort_value))}"
+                f"Got max absolute difference of: {np.amax(np.abs(ref_value - ort_value))} for "
+                f"{ref_value[bad_idxes]} vs {ort_value[bad_idxes]}"
```
Comment (lewtun) [inline src/transformers/onnx/convert.py:446]:
```suggestion
                f"{ref_value[bad_indices]} vs {ort_value[bad_indices]}"
```

**Review by lewtun — APPROVED** [review]
Thanks for the fix with the tolerance values for MarianMT exports @kventinel !

Can you please confirm that the slow tests pass with your change on `convert.py`, ie run this:

```
RUN_SLOW=1 pytest tests/onnx/test_onnx_v2.py -k "marian"
```

**Comment by kventinel** [comment]
> Can you please confirm that the slow tests pass with your change on convert.py, ie run this

6 passed, 426 deselected, 35 warnings

**Review by lewtun — APPROVED** [review]
Thanks for confirming the slow tests pass @kventinel 🤗 !

This PR LGTM, so gently pinging @sgugger for final approval

**Review by sgugger — APPROVED** [review]
Thanks for the fixes!

**Pull request merged**